src/Castle.Windsor.Tests/Lifestyle/ScopedLifestyleExplicitAndMultipleThreadsTestCase.cs

```diff
@@ -96,14 +95,12 @@ public void Context_is_passed_onto_the_next_thread_TPL()
 				var initialThreadId = Thread.CurrentThread.ManagedThreadId;
 				var task = Task.Factory.StartNew(() =>
 				{
-					Assert.AreNotEqual(Thread.CurrentThread.ManagedThreadId, initialThreadId);
 					instanceFromOtherThread = Container.Resolve<A>();
```

Comment:
I found a problem on my machine where if I debugged the test it would pass. Running it normally would make it fail. This test does not have proper task error handling so it swallows the real exception. On the off chance I did notice that the managed thread id was the same for both the task and the running test. Looks like the task ran on the same thread. The assert was kicking in and then squelching the real error. If netcoreapp1.0 is optimising TPL to run on the same thread, should we be asserting against it?

Reply:
@jonorossi ^^

Reply:
Will mess with it a bit more and report back. Curiously, notice how when I added a9f5f76 which had nothing to do with this test it passed again? I expect this is a flakey test that will only fail randomly every now and then.

Reply:
@jonorossi - This is further evidence of the flappy test which asserts on random integer. 

Please see this random failure when I updated markdown docs: https://ci.appveyor.com/project/castleproject/windsor/build/0.0.350#L321

This has got to be the managed thread id assertion! :)

Reply:
Boom! It passes! https://ci.appveyor.com/project/castleproject/windsor/build/0.0.351#L305

Reply:
Rubbish assertion, how MS optimise their threading have clearly changed in netcoreapp1.0.

Reply:
Jesus christ! Another flappy test.

https://ci.appveyor.com/project/castleproject/windsor/build/0.0.352#L736

Reply:
Passing again? 

https://ci.appveyor.com/project/castleproject/windsor/build/0.0.353#L231

Reply:
Yep, let's get rid of the assert.

With regards to the WCF test that failed, I've noticed in the past if the build machine is heavily loaded you can sometimes get failing tests like that, yep flakey.